features/FEATURE_LWIP/lwip-interface/lwip-eth/arch/TARGET_STM/TARGET_STM32F4/TARGET_MBED_CONNECT_ODIN/stm32f4_eth_init.c

```diff
@@ -0,0 +1,143 @@
+#include <string.h>
+#include "stm32f4xx_hal.h"
+#include "mbed_toolchain.h"
+
+#define C029_OTP_START_ADDRESS           (0x1FFF7800U)
+#define C029_OTP_END_ADDRESS             (C029_OTP_START_ADDRESS + (16*32))
+#define C029_MAC_ETHERNET_ID             (3)
+
+typedef MBED_PACKED(struct) C029_OTP_Header {
+    uint8_t id;
+    uint8_t len;
+    uint8_t data[];
+} C029_OTP_Header;
+
+static int _macRetrieved = 0;
+static char _macAddr[6] = { 0x02, 0x02, 0xF7, 0xF0, 0x00, 0x00 };
+
+static C029_OTP_Header *increment(C029_OTP_Header *pTemp)
+{
+    uint8_t len = 0;
+    uint8_t id = 0;
+    uint8_t *p = (uint8_t*)pTemp;
+
+    memcpy((void*)&id, (void*)pTemp, 1);
+
+    if (id == 0xFF){
+        p++;
```

Comment:
p++ is done in both parts of the conditional. Would it be better to write as :
```
p++;
if (id != 0xFF){
    memcpy((void*)&len, (void*)p++, 1);
    p += len;
}
```
??